### DISCUSSION TITLE
VXI11 Interfacing Question

### DISCUSSION THREAD

**Discussion by SBresler** [pymeasure/pymeasure#686] [General]
Hi,

I am using this package to write some drivers for some old SRS stuff.

The SG380 is one of my targets to write a good driver for.

I mostly want all of my stuff to use Ethernet > usb > GPIB > serial in terms of priority. Really everything is ultimately going to be ethernet. 

For the SG380, the manual states that The VXI11 interface has the ability to return to local remotely build into the interface. In other words, every time you want to press physical buttons on the device, you need to press shift+3 on the actual device to get back to local, unless you know how the VXI11 interface should handle it. This is a minor problem with 1 SG380, but I have... more than one. So the problem propagates. 

After digging through a bunch of stuff, I figured out that in python-VXI11 there is a command called local() that does this. It's not instrument specific. It's like the *IDN? command, but even more fundamental to the VXI11 interface. It's similar to the remote/local line on gpib I think  - it just doesn't have its own line because it's... not GPIB. 

I don't think this is in pymeasure, but it should be, and it should be at the lowest possible level (the adapter module I think). 

So my question is: I have never submitted a pull request to github and I want to implement my idea. Can someone help me?

I have already done all of this on my computer and it all works.... i just really don't want anyone to ever have to remember to push shift and then 3 its cruel.

**Comment by bilderbuchi** [comment]
Sure. If it already exists in python-vxi11 already has a function, it sounds straightforward enough. Normally we would discuss the needs and design first, then make a PR, but as you already have code, we could start with that and modify as needed.
If you're unsure about the workflow,  have you seen the Contributing guide in the docs? https://pymeasure.readthedocs.io/en/latest/dev/contribute.html
